test/csuite/timestamp_abort/main.c

```diff
@@ -340,6 +363,218 @@ set_flush_tier_delay(WT_RAND_STATE *rnd)
+
+        /* Process ranges within each file. */
+        testutil_check(__wt_snprintf(buf, sizeof(buf), "incremental=(file=%s)", filename));
+        testutil_check(session->open_cursor(session, NULL, cursor, buf, &file_cursor));
+
+        first_range = true;
+        rfd = wfd = 0;
+        while ((ret = file_cursor->next(file_cursor)) == 0) {
+            testutil_check(file_cursor->get_key(file_cursor, &offset, &size, &type));
+            testutil_assert(type == WT_BACKUP_FILE || type == WT_BACKUP_RANGE);
+
+            if (type == WT_BACKUP_RANGE) {
+
+                /*
+                 * If this is the first range, copy the base file from the source backup. Then open
+                 * the source and target files to set up the rest of the copy.
+                 */
+                if (first_range) {
+                    first_range = false;
+                    testutil_system(
+                      "cp %s/%s %s/%s", src_backup_home, filename, backup_home, filename);
+
+                    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", WT_HOME_DIR, filename));
+                    testutil_assert((rfd = open(buf, O_RDONLY, 0666)) >= 0);
+
+                    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/%s", backup_home, filename));
+                    testutil_assert((wfd = open(buf, O_WRONLY | O_CREAT, 0666)) >= 0);
+                }
+
+                /* Copy the range. */
+                tmp = dcalloc(1, size);
+                rdsize = pread(rfd, tmp, (size_t)size, (wt_off_t)offset);
+                testutil_assert(rdsize >= 0);
+                testutil_assert(pwrite(wfd, tmp, (size_t)rdsize, (wt_off_t)offset) == rdsize);
+                free(tmp);
+
+                nranges++;
+            } else {
+
+                /* We are supposed to do the full file copy. */
+                testutil_assert(first_range);
+                first_range = false;
+                testutil_system("cp %s/%s %s/%s", WT_HOME_DIR, filename, backup_home, filename);
+            }
+        }
+
+        if (rfd > 0)
+            testutil_assert(close(rfd) == 0);
+        if (wfd > 0)
+            testutil_assert(close(wfd) == 0);
+
+        testutil_assert(ret == WT_NOTFOUND);
+        testutil_check(file_cursor->close(file_cursor));
+    }
+    testutil_assert(ret == WT_NOTFOUND);
+
+    /* Cleanup */
+    testutil_check(cursor->close(cursor));
+    testutil_check(session->close(session, NULL));
+
+    /* Remember that the backup finished successfully. */
+    testutil_check(__wt_snprintf(buf, sizeof(buf), "%s/done", backup_home));
+    testutil_assert((wfd = open(buf, O_WRONLY | O_CREAT, 0666)) >= 0);
+    testutil_assert(close(wfd) == 0);
+
+    printf("Create backup: Files: %" PRId32 ", Ranges: %" PRId32 "\n", nfiles, nranges);
+}
+
+/*
+ * backup_delete_old_backups --
+ *     Delete old backups, keeping just a few recent ones, so that we don't take too much space for
+ *     no good reason.
```

Comment:
I don't understand how we have many backups. Can you add a comment somewhere to explain the strategy and algorithm for adding backups here?